tsdb/engine/tsm1/engine.go

```diff
@@ -167,17 +177,19 @@ func (e *DevEngine) compact() {
 		compact := segments[:n]
 
 		start := time.Now()
-		files, err := e.Compactor.Compact(compact)
+		files, err := e.Compactor.Compact(compact.Names())
 		if err != nil {
 			e.logger.Printf("error compacting WAL segments: %v", err)
 		}
 
 		// TODO: this is stubbed out but would be the place to replace files in the
 		// file store with the new compacted versions.
-		e.replaceFiles(files, compact)
+		e.replaceFiles(files, compact.Names())
 
-		// TODO: if replacement succeeds, we'd update the cache with the latest checkpoint.
-		// e.Cache.SetCheckpoint(...)
+		// Inform cache data may be evicted.
+		for _, id := range compact.IDs() {
```

Comment:
oh never mind, set checkpoint doesn't actually do what I was thinking. It would be better to rename `SetCheckpoint` to something like `PurgeCacheOfSegment`